nemo_skills/inference/eval/bfcl_web_search.py

```
@@ -0,0 +1,284 @@
+        Returns:
+            list: A list of search result dictionaries, each containing information such as:
+            - 'title' (str): The title of the search result.
+            - 'href' (str): The URL of the search result.
+            - 'body' (str): A brief description or snippet from the search result.
+        """
+        backoff = 10  # initial back-off in seconds
+
+        # Infinite retry loop with exponential backoff
+        while True:
+            try:
+                wait_time = backoff + random.uniform(0, backoff)
```

Ruff (0.14.0)

Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)